File: 6 Sorting/Triangle/triangle.js
function solution(A)
{
  A.sort((a, b) => a - b);

  for (let x = 0; x < (A.length-2); x++) {
    const p = A[x];
    const q = A[x+1];
    const r = A[x+2];

    if ((0 <= p < q < r < A.length) && (p + q) > r && (q + r) > p && (r + p) > q)
      return 1
    
  }
  return 0;
}

//console.log(solution([10,2,5,1,8,20]));
console.log(solution([10, 50, 5, 1]));

File: 6 Sorting/Triangle/Triangle.cpp
#include <algorithm>

int solution(vector<int> &A) {
  if (A.size() < 3) {
    return 0;
  }
  sort(A.begin(), A.end());

  for (unsigned int x = 2; x < A.size(); x++) {
    if ((long long)A[x - 2] + A[x - 1] > A[x]) return true;
  }
  return 0;
}
